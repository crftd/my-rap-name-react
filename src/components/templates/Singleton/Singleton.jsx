import React from 'react';
import styled from 'styled-components';

const Background = styled.div`
  height: 100vh;
  background: url(${props => props.backgroundUrl}) repeat 50% 50%;
`;

const FlexContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

const MainWrapper = styled.main`
  width: 80%;
  max-width: 600px;
  padding: 20px;
  background-color: rgba(255, 255, 255, 0.75);
  border-radius: 20px;
`;

export default ({ background, children }) => (
  <Background backgroundUrl={background}>
    <FlexContainer>
      <MainWrapper>{children}</MainWrapper>
    </FlexContainer>
  </Background>
);
