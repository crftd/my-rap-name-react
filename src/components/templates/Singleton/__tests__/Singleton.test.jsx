import React from 'react';
import { shallow } from 'enzyme';
import Singleton from '../Singleton.jsx';

describe('[Snapshot] Singleton', () => {
  it('should render', () => {
    // Arrange

    // Act
    const actualTree = shallow(<Singleton />);

    // Assert
    expect(actualTree).toMatchSnapshot();
  });
});
