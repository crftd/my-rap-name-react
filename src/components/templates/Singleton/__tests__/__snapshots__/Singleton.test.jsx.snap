// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] Singleton should render 1`] = `
<styled.div>
  <styled.div>
    <styled.main />
  </styled.div>
</styled.div>
`;
